### DISCUSSION TITLE
v0.3.13

### DISCUSSION THREAD

**Discussion by patrickgold** [florisboard/florisboard#1190] [Announcements]
## Added

- Integrated system spell checker service (#1052)
  You can find the spell checker by going to Settings > Typing > Spell checker.
  There are no language restrictions, dictionaries have to be imported manually.
  See in-app for more details.
- New audio and haptic feedback preferences (#1142)
  With this also the internal logic of audio/haptic feedback
  has been improved and some bugs have been fixed related
  to this feature. Make sure to check them out in Settings >
  Keyboard > Sounds & vibration
- Autofill inline support for devices running Android 11+ (#163)
- Add Japanese JIS layout (#140/#1109, thanks @Waelwindows)
- Add Halmak keyboard layout (#982, thanks @dessalines)
- Add Lithuanian layout / popup mapping (#1092)
- Add Neo2 layout (#527, thanks @ostrya)
- Alternative numpad with PC arrangement of digits (#1132)
- Indic Numerals (#940, thanks @yashpalgoyal1304)
- Basic support for direct boot mode (#1021, thanks @empratyush)
- Default system subtype for proper display in system settings
- Version number can now be copied by long-pressing in the about screen (#881, thanks @Luensche)
- Multiple symbol popup merging (#606, thanks @ostrya)
- Devtools override toggle for word suggestions heap size limit restriction (#1178)

## Changed

- Word suggestions are now only loading if sufficient memory is available. This
  restriction will be lifted again in 0.3.15 with the new word suggestion feature.
- Rework EditorInstance cached input and word detection logic
- Update turkish layout and popups to fix the capital I with dot bug (#1045)
- Reworked internal key data logic to make it easier to expand on (#1091)
- Some popups for Kurdish (#977, thanks @GoRaN909)
- Replaced none Kurmanci character "ı" by the correct one latin letter "i" (#1063, thanks @GoRaN909)
- (Internal) ICU dependency build scripts / source code submodule moved to
  separate repository as prep for F-Droid build recipe

## Improved

- Adjust Arabic popups for main forms and remove nums (#1087, thanks @Waelwindows)
- Remove popups not related to Turkish in Turkish layouts (#909, thanks @dvrnynr)
- Clipboard and glide typing performance (#1011, thanks @X-yl)
- Keyboard rendering and performance (#992)
- Duplicates in the clipboard view are now discarded (#991, thanks @Luensche)
- Keyboard view rendering performance (#966)
- Rapid-fire performance delays (#883)

## Fixed

- Gestures are now working properly on all screen densities (#1086)
- Fix delete words precisely / delete current word swipe
  - Fix crashes: #1130, #1076, #1065, #1014
  - Fix inconsistent selection: #546, #547
  - Fix deletion does not work: #1072
- Fix Smartbar Cut/Copy buttons not copying long texts correctly (#1146)
- Fix repeating keys (e.g. delete) not stopping sometimes (#986)
- Fix 5+1 keyboard layout bug (#1100)
- Cut off hints in borderless themes (#1049)
- View invalidation from incorrect thread (#1118)
- Crash when clearing clipboard containing image (#1124, thanks @X-yl)
- Crash after rebooting in clipboard manager (#1097)
- Double dotted circle for combining diacritical marks (#1089)
- Crash in Settings when layout is selected but not installed
- Fix theme editor preview looking distorted (#1136)
- Fix popup merge bug for group assigned keys (#1028)
- Fix candidates view showing for raw editors
- Fix NaN crash in layouting process (#1018)
- Smartbar not showing sometimes (#987)
- Keys not registering correctly, aka "being eaten" (#953)
- Keys not showing a shadow (#901, #921)
- Emoticon layout missing (#950)
- Crash when using delete left swipe in raw editors (#967)
- Devanagari Codes (#945, thanks @yashpalgoyal1304)
- Getting composer from name (#063, thanks @Hayleia)
- Crash loops from occurring after a crash (#910)
- User dictionary max size (#898)
- Space bar text too large (#862)
- Smartbar number row bugs in password fields (#905)
- One-handed panel not correctly measuring sometimes (#896)
- FlorisBoard Settings not accessible from Language & Input screen (#366)
- FlorisBoard not included in the keyboard switch cycle (#890)
- Space bar gestures for non-repeating actions (#886)
- Duplicate suggestions (#602)
- Hebrew layout flipped (#897, thanks @debnone)

Important: In this release the input IME ID of FlorisBoard changes, this may requires you to re-enable
FlorisBoard. This is a one-time action and after this update it won't happen again.

<hr /><em>This discussion was created from the release <a href='https://github.com/florisboard/florisboard/releases/tag/v0.3.13'>v0.3.13</a>.</em>

**Comment by paritosh9199** [comment]
Awesome work @patrickgold !!

**Reply by patrickgold** [reply]
Thanks!

**Comment by GoRaN9O** [comment]
You are amazing thank you for you work 💙

**Comment by castrik** [comment]
keyboard vibration doesn't work when holding delete key or when swiping on spacebar to navigate between words what used to happen was, if I swipe on spacebar, it would vibrate whener it switched to another letter or space or symbol, etc. now its gone same used to happen with delete key, there is only one initial vibration now

**Reply by patrickgold** [reply]
What are your vibration-related settings?

**Reply by castrik** [reply]
every setting enabled, vibration strength at 100%, duration at 3 ms
I've even set it to trigger via system and delete key vibration and spacebar swipe vibration still doesn't work

**Reply by patrickgold** [reply]
Strange. What happens if just for testing you set the duration at 50ms?

**Reply by castrik** [reply]
I figured it out, it seems the vibration strength is set at a really low percent and only changes the vibration strength for typing, not delete and cursor control

I can barely feel a really small vibration when holding the delete key and cursor control via spacebar and setting the vibration control to whatever percent doesn't change it
I realised it after changing the vibration strength at 50% and it felt close to what the delete key and cursor control have now

**Reply by PadawineAll** [reply]
I have no keyboard vibrations at all. Despite all parameter are on.

**Comment by IamSingh01** [comment]
Can't enable the incline suggestion and the clipboard history simultaneously.

**Reply by patrickgold** [reply]
You can enable word suggestions and the clipboard history, but sadly currently not the clipboard row in the Smartbar. This is planned however for a future release, so at some point when the Smartbar gets its refresh this will be possible.

**Comment by paritosh9199** [comment]
Hey @patrickgold When is the new beta coming out and where are we on fully fledged spellcheck and next word prediction?
BTW:  Really liking how this keyboard is turning out. Thanks for all the hardwork.

**Reply by patrickgold** [reply]
A new beta release will be coming out this weekend. The reason why it took so long is that I rebuilt the keyboard UI and logic internally so it fits the backend logic of the new preferences, and the old keyboard logic had some serious issues and would've caused a bit of trouble if not replaced.

As for next word predictions: this is something not covered by 0.3.14, but by 0.3.15/16.

Hope this helps!

**Comment by vishwanath16** [comment]
The glide typing isn't working right now.

I'm using Oneplus 9R  phone.

I've attached the screenshot for better understanding of the issue.

https://user-images.githubusercontent.com/31544005/143876362-9adcdae5-605d-4c03-9c4a-16dfe3b423da.mp4

**Reply by patrickgold** [reply]
Which FlorisBoard version are you on?

**Reply by vishwanath16** [reply]
0.3.13

![Screenshot_20211129-223055.jpg](https://user-images.githubusercontent.com/31544005/143910936-0442482b-93e6-4910-8fe0-748276be93a4.jpg)

**Reply by No1-found** [reply]
Try clearing app data and cache and try again,it worked for me.

**Comment by GaryPotthead1881** [comment]
Can't change size anymore with the new update and also swipe gestures aren't working.
Is there any solution known?

Systeminfo:
Xiaomi Mi 8
LineageOS 18.1
Android 11

**Reply by patrickgold** [reply]
> Can't change size anymore with the new update 

Size of what?

> also swipe gestures aren't working.

Maybe you have gesture typing enabled, which prevents swipe up/down/left/right but currently does nothing (will return in beta08)

**Reply by GaryPotthead1881** [reply]
> Size of what?

Sorry, forgot to note, I couldn't change the size of the board itself and also of the font.

> Maybe you have gesture typing enabled, which prevents swipe up/down/left/right but currently does nothing (will return in beta08)

Gestures were enabled, the board just wouldn't react to any of them.

Just tried to reinstall the board.
First time, total blackscreen, the board didn't want to load totaly. So it did second and third. Just on 4th time it installed completly and works fine again.

Thank you for your fast reply.
